app/models/mixins/authentication_mixin.rb

```diff
@@ -268,13 +268,12 @@ def authentication_check(*args)
 
   def authentication_check_no_validation(type, options)
     header  = "type: [#{type.inspect}] for [#{id}] [#{name}]"
-    verify_args = self.kind_of?(Host) ? [type, options] : type
     status, details =
       if self.missing_credentials?(type)
         [:incomplete, "Missing credentials"]
       else
         begin
-          verify_credentials(*verify_args) ? [:valid, ""] : [:invalid, "Unknown reason"]
+          verify_credentials(type, options) ? [:valid, ""] : [:invalid, "Unknown reason"]
```

Comment:
@durandom Do you know if there are any cases where self is not a Host and options should not be passed for some specific reason? Because it seems odd that it was originally special cased for Host in the first place.

If not, this change seems fine.